### PR TITLE
Move checkRequiredFields to deployer.New and mark all fields required

### LINKED ISSUE #954: Code cleanup: Refactor config.CheckRequiredField
**What happened**

The PR: https://github.com/Azure/ARO-RP/pull/848 introduced checkRequiredField method for config struct. This should belong to the receiver of config. 

**How it should look like**

CheckRequiredField should be part of the receiver New.

### PR DESCRIPTION
**Pull request by petrkotas**
### Which issue this PR addresses:

Fixes #954

### What this PR does / why we need it:

TL;DR: Move the checkRequiredField to the `deployer.New` which is a more logical place for this code.

The checkRequiredField is left in the `config.go` source, but is left unexposed.
This enables easier reading as the code is linked to work on the config struct.
Call to the `checkRequiredField` is done in the `deployer.New` as the first step,
and fails when the config does not meet the required field criteria.

This PR also sets all fields as required, as this is the current reality.

Signed-off-by: Petr Kotas <pkotas@redhat.com>

### Test plan for issue:

Tests already in place for existing code.

### Is there any documentation that needs to be updated for this PR?

None needed.

### CODE REVIEW

**Review by asalkeld — APPROVED** [review]

File: pkg/deploy/config.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func mergeConfig(primary, secondary *Configuration) (*Configuration, error) {
 
 // CheckRequiredFields validates configuration whether it provides required fields.
 // Config is invalid if required fields are not provided.
-func (conf *RPConfig) CheckRequiredFields() error {
+func checkRequiredFields(conf RPConfig) error {
```
Comment (jim-minter) [inline pkg/deploy/config.go:103]:
```suggestion
func (conf *RPConfig) validate() error {
```

File: pkg/deploy/config_test.go [hunk 2]
```diff
@@ -128,10 +147,32 @@ func TestConfigRequiredValues(t *testing.T) {
...
 				Configuration: &Configuration{
-					ACRResourceID:    &ACRResourceID,
-					AdminAPICABundle: &AdminAPICABundle,
-					ExtraCosmosDBIPs: []string{ExtraCosmosDBIPs},
-					MDMFrontendURL:   &MDMFrontendURL,
+					ACRResourceID:                      &ACRResourceID,
+					AdminAPICABundle:                   &AdminAPICABundle,
+					ExtraCosmosDBIPs:                   []string{ExtraCosmosDBIPs},
+					MDMFrontendURL:                     &MDMFrontendURL,
+					ACRReplicaDisabled:                 &tr,
```
Comment (jim-minter) [inline pkg/deploy/config_test.go:154]: use `to.BoolPtr(true)`

**Review by jim-minter — DISMISSED** [review]
This PR fails to validate the current int and prod RP configs.

**Comment by petrkotas** [comment]
I have updated the PR according to the recommendations. Thanks for all the pointers.
I had to remove: 

* ACRReplicaDisabled
* FPServerCertCommonName
* RPMode

from the required fields, as they are not present either in INT or PROD config. Since the config runs fine, I expect them to not be required.

**Review by mjudeikis — APPROVED** [review]
